x-pack/plugins/alerting/server/maintenance_window_client/methods/archive.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import Boom from '@hapi/boom';
+import {
+  generateMaintenanceWindowEvents,
+  mergeEvents,
+} from '../generate_maintenance_window_events';
+import { getMaintenanceWindowFromRaw } from '../get_maintenance_window_from_raw';
+import {
+  MaintenanceWindowSOAttributes,
+  MaintenanceWindow,
+  MAINTENANCE_WINDOW_SAVED_OBJECT_TYPE,
+  MaintenanceWindowClientContext,
+} from '../../../common';
+import { retryIfConflicts } from '../../lib/retry_if_conflicts';
+
+export interface ArchiveParams {
+  id: string;
+  archive: boolean;
+}
+
+const getArchivedExpirationDate = (shouldArchive: boolean) => {
+  if (shouldArchive) {
+    return new Date().toISOString();
+  }
+  return moment.utc().add(1, 'year').toISOString();
+};
+
+export async function archive(
+  context: MaintenanceWindowClientContext,
+  params: ArchiveParams
+): Promise<MaintenanceWindow> {
+  return await retryIfConflicts(
```

Comment:
I am wondering if we really need to do that or just use the saved object options `retryOnConflict`. They already doing it for us what do you think?